include/utils/MastodonUtils.h

```diff
@@ -40,6 +42,83 @@ std::vector<std::vector<Real>> responseSpectrum(const Real & freq_start,
 std::vector<std::vector<Real>> regularize(const std::vector<Real> & history_acc,
                                           const std::vector<Real> & history_time,
                                           const Real & reg_dt);
+
+/**
+ * Function to check if all the vectors in a vector are of equal size
+ */
+bool checkVectorsUnequalSize(std::vector<std::vector<Real>> & vectors);
+
+/**
+ * Function to check if two vectors are equal
+ */
+bool
+areVectorsEqual(std::vector<Real> & vector1, std::vector<Real> & vector2, Real percent_error = 0.0);
+
+/**
+ * Function to check if the vector contains any non-positive elements
+ */
+bool isVectorNonPositive(const std::vector<Real> & vector);
+
+/**
+ * Function to calculate the mean of the elements in a vector
+ */
+Real mean(std::vector<Real> & vector);
+
+/**
+ * Function to calculate the median of the elements in a vector
+ */
+Real median(std::vector<Real> & vector, const std::string & interpolation = "linear");
+
+/**
+ * Function to calculate the percentile value of the elements in a vector
+ */
+Real percentile(std::vector<Real> & vector,
+                const Real & percent,
+                const std::string & interpolation = "linear");
+
+/**
+ * Function to calculate the standard deviation of the elements in a vector
+ */
+Real sigma(std::vector<Real> & vector);
+
+/**
+ * Function to calculate the lognormal standard deviation of the elements in a vector
+ */
+Real beta(std::vector<Real> & vector);
+
+/**
+ * Function template to calculate the probability that one random variable is greater
+ * than another random variable
+ */
+template <typename T>
+Real
+greaterProb(T & demand_distribution, T & capacity_distribution)
+{
```

Comment:
The syntax looks okay for this function. I don't know enough probability to know whether the math is correct.

Reply:
The function has a unit test for an obvious case (result should be 0.5 for two identical distributions).

Reply:
Call it "greaterProbability". Also, why is this a template? What types are the distributions that are being passed in, if they have a common base you could use the base type.

Reply:
@aeslaughter we talked about this a while ago. You mentioned that Boost doesn't have a base class for all distributions..